cmd/gb-list/main.go

```diff
@@ -47,9 +56,27 @@ func main() {
 	if err != nil {
 		gb.Fatalf("unable to resolve: %v", err)
 	}
-	for _, pkg := range pkgs {
-		if err := tmpl.Execute(os.Stdout, pkg); err != nil {
-			gb.Fatalf("unable to execute template: %v", err)
+
+	if jsonOutput {
+		views := make([]*PackageView, 0, len(pkgs))
+		for _, pkg := range pkgs {
+			views = append(views, NewPackageView(pkg))
+		}
+		encoded, err := json.MarshalIndent(views, " ", "  ")
```

Comment:
I forgot that go list even spoke json. If you hadn't already written that
code, I would have said not to work about it.

On Thu, 14 May 2015 08:31 Mario Kostelac <notifications@github.com> wrote:

> In cmd/gb-list/main.go
> <https://github.com/constabulary/gb/pull/77#discussion_r30282829>:
>
> > @@ -47,9 +56,27 @@ func main() {
> >  	if err != nil {
> >  		gb.Fatalf("unable to resolve: %v", err)
> >  	}
> > -	for _, pkg := range pkgs {
> > -		if err := tmpl.Execute(os.Stdout, pkg); err != nil {
> > -			gb.Fatalf("unable to execute template: %v", err)
> > +
> > +	if jsonOutput {
> > +		views := make([]*PackageView, 0, len(pkgs))
> > +		for _, pkg := range pkgs {
> > +			views = append(views, NewPackageView(pkg))
> > +		}
> > +		encoded, err := json.MarshalIndent(views, " ", "  ")
>
> If it is just for machines, indentation is not important, I agree.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/77/files#r30282829>.
>